src/components/clusters/newClusterModal.tsx

```diff
@@ -73,18 +57,30 @@ export const NewClusterModal: React.FC<NewClusterModalProps> = ({ closeModal })
 
   const validationSchema = React.useCallback(
     () =>
-      Yup.object().shape({
-        name: getUniqueNameValidationSchema(clusterNames).concat(nameValidationSchema),
+      Yup.object({
+        name: nameValidationSchema,
         openshiftVersion: Yup.string().required('Required'),
       }),
-    [clusterNames],
+    [],
   );
 
   const handleSubmit = async (
     values: ClusterCreateParams,
     formikActions: FormikHelpers<ClusterCreateParams>,
   ) => {
     formikActions.setStatus({ error: null });
+
+    // async validation for cluster name - run only on submit
+    try {
```

Comment:
What about running it twice? Get list of clusters when opening the page, keep validating against it and then (if really needed) refresh when submitting.

Reply:
In a separate change we could add a field level validation for the name field to do the async validation onBlur.